modules/sdk-core/src/bitgo/bitgoBase.ts

```diff
@@ -0,0 +1,43 @@
+import * as superagent from 'superagent';
+import { EnvironmentName } from './environments';
+
+export interface BitGoRequest<ResultType = any> extends superagent.SuperAgentRequest {
+  result: (optionalField?: string) => Promise<ResultType>;
+}
+
+export interface DecryptOptions {
+  input: string;
+  password?: string;
+}
+
+export interface EncryptOptions {
+  input: string;
+  password?: string;
+}
+
+export interface GetSharingKeyOptions {
+  email: string;
+}
+
+export interface IRequestTracer {
+  inc(): void;
+  toString(): string;
+}
+
+export interface BitGoBase {
+  coin(coinName: string): any; // need to change it to BaseCoin once it's moved to @bitgo/sdk-core
```

Comment:
we shouldn't use `any` here. We need to define the `BaseCoin` interface, which requires defining the other interfaces. 🐔  <-> 🥚

Reply:
I agree with you on that. The reason I used `any` there was to resolve an issue in one of the classes in `bitgo` module. Also, `BaseCoin` is still in `bitgo` module so I did not want to import `BaseCoin` from `bitgo` since it would create a circular dependency between `bitgo` and `sdk-core`.